runtime/compiler/control/CompilationThread.cpp

```diff
@@ -2550,6 +2552,144 @@ TR::CompilationInfoPerThread* TR::CompilationInfo::getCompInfoForThread(J9VMThre
+   _compThreadActivationThresholdsonStarvation = static_cast<int32_t *>(jitPersistentAlloc((numTotalCompThreads + 1) * sizeof(int32_t)));
+
+   _arrayOfCompilationInfoPerThread = static_cast<TR::CompilationInfoPerThread **>(jitPersistentAlloc(numTotalCompThreads * sizeof(TR::CompilationInfoPerThread *)));
+
+   if (_compThreadActivationThresholds &&
+       _compThreadSuspensionThresholds &&
+       _compThreadActivationThresholdsonStarvation &&
+       _arrayOfCompilationInfoPerThread)
+      {
+      // How to read it: the queueWeight has to be over 100 to activate second comp thread
+      //                 the queueWeight has to be below 10 to suspend second comp thread
+      // For example:
+      // compThreadActivationThresholds[MAX_TOTAL_COMP_THREADS+1] = {-1, 100, 200, 300, 400, 500, 600, 700, 800};
+      // compThreadSuspensionThresholds[MAX_TOTAL_COMP_THREADS+1] = {-1,  -1,  10, 110, 210, 310, 410, 510, 610};
+      // compThreadActivationThresholdsonStarvation[MAX_TOTAL_COMP_THREADS+1] = {-1, 800, 1600, 3200, 6400, 12800, 19200, 25600, 32000};
+      _compThreadActivationThresholds[0] = -1;
+      _compThreadActivationThresholds[1] = 100;
+      _compThreadActivationThresholds[2] = 200;
+
+      _compThreadSuspensionThresholds[0] = -1;
+      _compThreadSuspensionThresholds[1] = -1;
+      _compThreadSuspensionThresholds[2] = 10;
+
+      for (int32_t i=3; i<(numTotalCompThreads+1); ++i)
+         {
+         _compThreadActivationThresholds[i] = _compThreadActivationThresholds[i-1] + 100;
+         _compThreadSuspensionThresholds[i] = _compThreadSuspensionThresholds[i-1] + 100;
+         }
+
+      _compThreadActivationThresholdsonStarvation[0] = -1;
+      _compThreadActivationThresholdsonStarvation[1] = 800;
+
+      for (int32_t i=2; i<(numTotalCompThreads+1); ++i)
+         {
+         if (_compThreadActivationThresholdsonStarvation[i-1] < 12800)
+            {
+            _compThreadActivationThresholdsonStarvation[i] = _compThreadActivationThresholdsonStarvation[i-1] * 2;
+            }
+         else
+            {
+            _compThreadActivationThresholdsonStarvation[i] = _compThreadActivationThresholdsonStarvation[i-1] + 6400;
+            }
+         }
+
+      for (int32_t i=0; i<numTotalCompThreads; ++i)
+         {
+         _arrayOfCompilationInfoPerThread[i] = NULL;
+         }
+      return true;
+      }
+   return false;
+   }
+
+void
+TR::CompilationInfo::freeAllCompilationThreads()
+   {
+   if (_compThreadActivationThresholds)
+      {
+      jitPersistentFree(_compThreadActivationThresholds);
+      }
+
+   if (_compThreadSuspensionThresholds)
+      {
+      jitPersistentFree(_compThreadSuspensionThresholds);
+      }
+
+   if (_compThreadActivationThresholdsonStarvation)
+      {
+      jitPersistentFree(_compThreadActivationThresholdsonStarvation);
+      }
+
+   if (_arrayOfCompilationInfoPerThread)
+      {
+      jitPersistentFree(_arrayOfCompilationInfoPerThread);
```

Comment:
Just calling jitPersistentFree is not enough because CompilationInfoPerThread has other dynamicallt allocated fields. You need to define a virtual destructor and call that. Since today we don't destroy those objects I would say to eliminate this jitPersistentFree and add a TODO comment plus open an issue to track it.